src/ripple/protocol/impl/tokens.cpp

```diff
@@ -266,7 +267,7 @@ decodeBase58Token (std::string const& s,
         return {};
 
     // The type must match.
-    if (type != static_cast<TokenType>(ret[0]))
+    if (type != safe_cast<TokenType>(static_cast<std::uint8_t>(ret[0])))
```

Comment:
Done except I retained the implicit conversion syntax from the return type of `decodeBase58` because I don't want to accidentally invoke an explicit conversion from `decodeBase58` should its return type change in the future.